src/Lykke.Service.FixGateway.Services/NewOrderRequestHandler.cs

```diff
@@ -0,0 +1,268 @@
+        {
+            if (response == null)
+            {
+                var exception = new InvalidOperationException("ME not available");
+                await _log.WriteErrorAsync(nameof(NewOrderRequestHandler), nameof(CheckResponseAndThrowIfNullAsync), exception);
+                throw exception;
+            }
+        }
+
+        private async Task<MarketOrderFeeModel> GetMarketOrderFeeAsync(string assetPairId, OrderAction orderAction)
+        {
+            var assetPair = await _assetsServiceWithCache.TryGetAssetPairAsync(assetPairId);
+            var fee = await _feeCalculatorClient.GetMarketOrderFees(_credentials.ClientId.ToString("D"), assetPairId, assetPair?.BaseAssetId, _mapper.Map<FeeCalculator.AutorestClient.Models.OrderAction>(orderAction));
+
+            return new MarketOrderFeeModel
+            {
+                Size = (double)fee.Amount,
+                SourceClientId = _credentials.ClientId.ToString("D"),
+                TargetClientId = _feeSettings.TargetClientId.Hft,
+                Type = (int)MarketOrderFeeType.CLIENT_FEE
+            };
+        }
+
+        private static Guid GenerateOrderId()
+        {
+            return Guid.NewGuid();
+        }
+
+        private async Task<bool> ValidateRequestAsync(Guid newOrderId, NewOrderSingle request)
+        {
+            var clOrdId = request.ClOrdID.Obj;
+            var oe = await _clientOrderIdProvider.CheckExistsAsync(clOrdId);
+            Message reject = null;
+            if (oe)
+            {
+                reject = CreateRejectResponse(newOrderId, request, OrdRejReason.DUPLICATE_ORDER);
+            }
+
+            var allSymbols = (await _assetsServiceWithCache.GetAllAssetPairsAsync(_tokenSource.Token)).Select(a => a.Id).ToHashSet();
+
+            if (!allSymbols.Contains(request.Symbol.Obj))
+            {
+                reject = CreateRejectResponse(newOrderId, request, OrdRejReason.UNKNOWN_SYMBOL);
+            }
+
+
+            if (!Guid.TryParse(request.Account.Obj, out var accId) || accId != _credentials.ClientId)
+            {
+                reject = CreateRejectResponse(newOrderId, request, OrdRejReason.UNKNOWN_ACCOUNT);
+            }
+
+            if (request.OrderQty.Obj <= 0
+                || !new[] { Side.BUY, Side.SELL }.Contains(request.Side.Obj)
+                || !new[] { OrdType.MARKET, OrdType.LIMIT }.Contains(request.OrdType.Obj)
+                || request.OrdType.Obj == OrdType.LIMIT && (!request.IsSetPrice() || request.Price.Obj <= 0m)
+                || !new[] { TimeInForce.GOOD_TILL_CANCEL, TimeInForce.FILL_OR_KILL }.Contains(request.TimeInForce.Obj)
+                || request.OrdType.Obj == OrdType.LIMIT && request.TimeInForce.Obj != TimeInForce.GOOD_TILL_CANCEL)
+            {
+                reject = CreateRejectResponse(newOrderId, request, OrdRejReason.UNSUPPORTED_ORDER_CHARACTERISTIC);
+            }
+
+
+
+            if (reject != null)
+            {
+                SendReject(reject);
```

Comment:
strange logic, wwe can rewrite variable many time and send last variand
mayme need dend after first detected and return